dell/csi-powerscale
# unresolvable hosts param support

# Description
Added support to add host to OneFS exports though existing export has some hosts which are unresolvable but doesn't stop adding new hosts to same exports.

This is parameterized support and can be benefited via "**values.yaml**" by enabling "**ignoreHosts**" param to "**true**".
**Default its set to "false" and maintain existing behavior of the Driver.**

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/534|

# Checklist:

- [x] Backward compatibility is not broken

# How Has This Been Tested?
Test scenarios covered : 

1. **ignoreHosts set to "true"** 
- Create a RWX volume --> pass
- Create a POD --> pass
- Add dummy export to the above created volume export directly on OneFS --> Added (manually)
- Now, create another POD which claims same volume --> Pass 
- Delete PODs one-by-one and check hosts entries are also removed from OneFS corresponding export --> Pass

2. **ignoreHosts set to "false"**
- Create a RWX volume --> pass
- Create a POD --> pass
- Add dummy export to the above created volume export directly on OneFS --> Added (manually)
- Now, create another POD which claims same volume --> Fail
- Delete PODs one-by-one and check hosts entries are also removed from OneFS corresponding export --> Fail
